src/sagas/index.js

```diff
@@ -139,13 +139,13 @@ export function subscribe(session) {
 
         function on_connection(args) {
             var connection = args[0];
-
+            const {listening, port_statuses, connected} = connection
```

Comment:
Will that work if sent object is 
```
{
    'listening': False
}
```
? Will it fail because of missing fields, or will it just assign None?

Reply:
No control assigned for the `listening` yet, electron won't fail whatever `listening` is; null/false etc.. 
What kind of reaction the GUI has to give if `listening` is negative @etam @jamuszyn ?

Reply:
The question was not about 'listening' value. The question was, if the object contains only 'listening' key, but no 'port_statuses' nor 'connected', what will happen with the corresponding variables?

Reply:
I think that implementation of any reactions should be a separate issue. This change is only about changing the value sent through RPC from a string to the source data. Let's keep it simple.

Reply:
@etam If any implementation will be possible, then yes It gonna be any other issue. But if not I will just skip the rest of the connection function steps and it will be the solution. And there won't be any fail, I think this is so far the optimal answer for your question.  I'm keeping this branch `in progress` until find the best approach with @jamuszyn.